client/components/community-translator/utils.js

```diff
@@ -0,0 +1,174 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import request from 'superagent';
+import { head, find, get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import userSettings from 'lib/user-settings';
+import { isMobile } from 'lib/viewport';
+import { GP_PROJECT, GP_BASE_URL, GP_PROJECT_TRANSLATION_SET_SLUGS } from './constants';
+import { canBeTranslated } from 'lib/i18n-utils';
+
+/**
+ * Checks whether the CT can be displayed, that is, if the chosen locale and device allow it
+ * @param {String} locale user's localeSlug
+ * @param {Object} localeVariant user's localeVariant slug (if any)
+ * @returns {Boolean} whether the CT can be displayed
+ */
+export function canDisplayCommunityTranslator(
+	locale = userSettings.getSetting( 'language' ),
+	localeVariant = userSettings.getSetting( 'locale_variant' )
+) {
+	// restrict mobile devices from translator for now while we refine touch interactions
+	if ( isMobile() ) {
+		return false;
+	}
+
+	// disable for locales with no official GP translation sets.
+	if ( ! locale || ! canBeTranslated( locale ) ) {
+		return false;
+	}
+
+	// likewise, disable for locale variants with no official GP translation sets
+	if ( localeVariant && ! canBeTranslated( localeVariant ) ) {
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * Checks whether the CT is enabled, that is, if
+ * 1) the user has chosen to enable it,
+ * 2) it can be displayed based on the user's language and device settings
+ * @returns {Bool} whether the CT should be enabled
+ */
+export function isCommunityTranslatorEnabled() {
+	if ( ! userSettings.getSettings() || ! userSettings.getOriginalSetting( 'enable_translator' ) ) {
+		return false;
+	}
+
+	if ( ! canDisplayCommunityTranslator() ) {
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * Sends the POST request
+ * @param {String} glotPressUrl API url
+ * @param {String} postFormData post data url param string
+ * @returns {Object} request object
+ */
+export function postRequest( glotPressUrl, postFormData ) {
+	return request
+		.post( glotPressUrl )
+		.withCredentials()
+		.send( postFormData )
+		.then( response => normalizeDetailsFromTranslationData( head( response.body ) ) )
```

Comment:
I found a weird bug, which for some reason, I didn't notice before.

The `-query-by-originals` and `-new` responses differ, so the lodash method `head()` used to parse it yields different results.

**-query-by-originals**
`[{"original_id":"225303","original":...`

**-new**
`{"225303":[{"id":"7309462","original_id":"225303...`

This will get it working until we can think of something more robust, unless you have a better idea:

`return normalizeDetailsFromTranslationData( head( flatMap( response.body ) ) )`